jni/src/common/Foreign/JNI/String.hs

```diff
@@ -66,7 +66,7 @@ toByteString (String bs) = BS.init bs
 -- a null-terminated buffer first.
 fromByteString :: ByteString -> String
 fromByteString bs
-  | BS.last bs == 0 = String bs
+  | BS.length bs > 0 && BS.last bs == 0 = String bs
   | otherwise = String (bs `BS.snoc` 0)
```

Comment:
There is `fromByteString` and there is `unsafeFromByteString`, both functions are unsafe in that if the given bytestring is not in modified UTF8, then the returned `JNI.String` would be invalid.

I don't know how to fix that. Maybe we could rename them to `unsafeFromByteString` and `unsafeFromNullTerminatedByteString`.

Reply:
I agree that these functions are improperly named. unsafeFromByteString and unsafeFromNullTerminatedByteString at least carry the meaning that none of them are safe, but I find it a bit too verbose.

As for fixing the underlying problem, the only solution I see would be to implement the switch from standard utf8 to modified utf8 ourselves. This probably deserve its own issue.